crates/prune/src/prune.rs

```diff
@@ -0,0 +1,158 @@
+//! Support for maintaining the state of the transaction pool
+
+use futures_util::{Stream, StreamExt};
+use reth_primitives::BlockNumber;
+use reth_provider::CanonStateNotification;
+use std::{
+    future::Future,
+    pin::Pin,
+    task::{ready, Context, Poll},
+};
+use tracing::debug;
+
+// TODO(alexey): use config field from https://github.com/paradigmxyz/reth/pull/3341
+/// Minimum pruning interval measured in blocks. All prune parts are checked and, if needed, pruned,
+/// when the chain advances by the specified number of blocks.
+pub const MIN_PRUNE_BLOCK_INTERVAL: u64 = 5;
+
+/// Pruning routine. Implements [Future] where the pruning logic happens.
+pub struct Pruner<St, Client> {
+    /// Stream of canonical state notifications. Pruning is triggered by new incoming
+    /// notifications.
+    canon_state_stream: St,
+    /// Database interaction client.
+    #[allow(dead_code)]
+    client: Client,
+    /// Maximum reorg depth. Used to determine the pruning target for parts that are needed during
+    /// the reorg, e.g. changesets.
+    #[allow(dead_code)]
+    max_reorg_depth: u64,
```

Comment:
I would like to make this explicit and rename this field so it is clear what it is used for.

for example `prune_depth`, 
because if we want to change where we want to prune, then we need to express this in terms of `reorg`

Reply:
agree, changed